WalletWasabi.Backend/Controllers/ArenaController.cs

```diff
@@ -0,0 +1,59 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.Backend.Controllers
+{
+	/// <summary>
+	/// To make batched requests.
+	/// </summary>
+	[Produces("application/json")]
+	[ApiController]
+	[Route("[controller]")]
+	public class ArenaController : ControllerBase
+	{
+		private IArenaRequestHandler handler;
+
+		public ArenaController(IArenaRequestHandler handler)
+		{
+			this.handler = handler;
+		}
+
+		public Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(ConnectionConfirmationRequest request)
+		{
+			return handler.ConfirmConnectionAsync(request);
+		}
+
+		[HttpPost("registerinput")]
+
+		public Task<InputRegistrationResponse> RegisterInputAsync(InputRegistrationRequest request)
+		{
+			return handler.RegisterInputAsync(request);
+		}
+
+		[HttpPost("registeroutput")]
+		public Task<OutputRegistrationResponse> RegisterOutputAsync(OutputRegistrationRequest request)
+		{
+			return handler.RegisterOutputAsync(request);
+		}
+
+		[HttpPost("reissuecredential")]
```

Comment:
That's not how you name HTTP endpoints. Never issue commands. You either `GET` resources or `POST` resources. Never issue commands. Never use verbs, always use nouns. Use `-` to separate words. Eg. `POST registerinput` -> `POST input-registration` or just `POST input` if it's obvious.

Reply:
how do you feel about HTTP methods like `PUT` and `DELETE` for fully RESTful style?

Reply:
I believe using rarer HTTP requests like `PUT` and `DELETE` is a not a good idea because some clients don't implement them. I can recall a similar scenario when I added a content field for a `GET` request and the client - probably Postman - that I used wasn't able to add content field for `GET` requests, only for `POST`s, even though content for `GET` is perfectly valid according to the HTTP protocol.